crates/exex/src/context.rs

```diff
@@ -41,4 +35,46 @@ pub struct ExExContext<Node: FullNodeComponents> {
     /// Once a an [`ExExNotification`] is sent over the channel, it is considered delivered by the
     /// node.
     pub notifications: Receiver<ExExNotification>,
+
+    /// node components
+    pub components: Node,
+}
+
+impl<Node: FullNodeComponents> NodeTypes for ExExContext<Node> {
+    type Primitives = Node::Primitives;
+    type Engine = Node::Engine;
+}
+
+impl<Node: FullNodeComponents> FullNodeTypes for ExExContext<Node> {
+    type DB = Node::DB;
+    type Provider = Node::Provider;
+}
+
+impl<Node: FullNodeComponents> FullNodeComponents for ExExContext<Node> {
```

Comment:
I think these are nice to have but we could also live without
the trait is important anyway, so this is fine